pandas/_libs/reduction.pyx

```diff
@@ -494,7 +494,8 @@ class InvalidApply(Exception):
 
 
 def apply_frame_axis0(object frame, object f, object names,
-                      ndarray[int64_t] starts, ndarray[int64_t] ends):
+                      ndarray[int64_t] starts, ndarray[int64_t] ends,
+                      skip_first_index = True):
```

Comment:
Would you please add a docstring, or at least clarify the role of ``skip_first_index``? (I can't get it)

Reply:
sorry. `skip_first_index` was to do debugging quicker with cython. `skip_first_index = True` is with the changes. `skip_first_index = False` is without. I'll remove it and make a commit.

Reply:
OK, no problem

Reply:
not sure why you need to add a parameter here? fast_apply (which calls this) is the first group operation, so by definition this is never called twice. The following code in pandas/core/groupby/ops.py *can* call the groups again, but that is only if the fast path has failed.